### PR TITLE
Charge, Cancel, Refund

### LINKED ISSUE #20: Implement transaction-refund-requested webhook

### LINKED ISSUE #24: Implement transaction-cancelation-requested webhook

### PR DESCRIPTION
**Pull request by mmiszy**
Fixes #23
Fixes #24
Fixes #20

### CODE REVIEW

File: src/modules/stripe/__recordings__/stripe-api_748119249/validateStripeKeys_2838756478/should-throw-error-if-publishable-key-is-invalid_3942585802/recording.har [hunk 1]
```diff
@@ -59,47 +59,40 @@
...
-                    "allow_redirects": "always",
-                    "enabled": true
-                  },
+                  "automatic_payment_methods": null,
                   "canceled_at": null,
                   "cancellation_reason": null,
-                  "capture_method": "automatic",
-                  "client_secret": "pi_3NKgQdEosEcNBN5m0mRPtkaO_secret_BfW3ubNBjevc23pXb1D5IYMyd",
+                  "capture_method": "manual",
+                  "client_secret": "pi_3NRGSfEosEcNBN5m0kByGEJD_secret_VVj8JJTJVmYvyJGtQkSTwRpMN",
```
Comment (witoszekdev) [inline src/modules/stripe/__recordings__/stripe-api_748119249/validateStripeKeys_2838756478/should-throw-error-if-publishable-key-is-invalid_3942585802/recording.har:66]: Is it ok to store `client_secret`?

File: src/modules/webhooks/__tests__/utils.ts [hunk 2]
```diff
@@ -471,3 +471,9 @@ export const createMockTransactionProcessSessionEvent: CreateMockAsyncFn<
     overrides,
   );
 };
+
+export const createMockStripeDataActionNotRequired = () => {
+  return {
+    // return_url: testEnv.TEST_SALEOR_API_URL + "checkout",
```
Comment (witoszekdev) [inline src/modules/webhooks/__tests__/utils.ts:477]: Why is it commented?

File: src/modules/webhooks/stripe-webhook.ts [hunk 3]
```diff
@@ -323,15 +323,88 @@ function stripeEventToPartialToTransactionEventReportMutationVariables(
...
+        return {
+          type: TransactionEventTypeEnum.RefundFailure,
+          message: stripeEvent.data.object.failure_reason,
+        };
+      case `pending`:
+        return { type: TransactionEventTypeEnum.RefundRequest };
+      case `succeeded`:
+        return { type: TransactionEventTypeEnum.RefundSuccess };
+      case `requires_action`:
+        return { type: TransactionEventTypeEnum.RefundRequest, message: `requires_action` };
```
Comment (witoszekdev) [inline src/modules/webhooks/stripe-webhook.ts:373]: Let's stick to double quotes `"`

File: src/modules/webhooks/transaction-initialize-session.test.ts [hunk 4]
```diff
@@ -24,10 +24,10 @@ describe(`TransactionInitializeSessionWebhookHandler`, () => {
           "x-stripe-client-user-agent",
         ],
       },
-      method: false,
-      body: false,
-      order: false,
-      url: false,
+      method: true,
+      body: true,
+      order: true,
```
Comment (witoszekdev) [inline src/modules/webhooks/transaction-initialize-session.test.ts:29]: Does the order really matter in recordings? 🤔

Comment (mmiszy) [reply src/modules/webhooks/transaction-initialize-session.test.ts:29]: You're right, it shouldn't

File: src/modules/webhooks/__tests__/utils.ts (hunk 2 above)
Comment (mmiszy) [reply src/modules/webhooks/__tests__/utils.ts:477]: Should be deleted

**Review by witoszekdev — APPROVED** [review]

**Pull request merged**